content/TruenasApps/StableApps/InstallNextCloudMedia.md

```diff
@@ -15,151 +15,284 @@ keywords:
-      This example uses *admin* as the user.
+{{< expand "Nextcloud Cron Jobs" "v" >}}
+NextCloud cron jobs only run while the app is running. If you stop the app, the cron job(s) do not run until you start the app again.
 
-   {{< trueimage src="/images/SCALE/Apps/InstallNextcloudConfigUsernamePassword.png" alt="Nextcloud Configuration Username" id="Nextcloud Configuration Username" >}}
+For more information on formatting and using cron jobs, see [Managing Cron Jobs]({{< relref "ManageCronJobsSCALE.md" >}}).
+{{< /expand >}}
 
-   TrueNAS populates **Host** with the IP address for your TrueNAS server and **Nextcloud data directory** populates with the correct path.
+The TrueNAS app is configured with all the required environment variables, but if you want to further customize the container, click **Add** to the right of **Additional Environment Variables** for each to enter the variable(s) and values(s).
+{{< /expand >}}
+Enter the network configuration settings. 
+Enter the default port, **30027**, in **WebUI Port**, or enter an available port number of your choice.
+See [Network Configuration](#network-configuration) below for more information on changing the default port.
+This port must match the one used in **Host** above.
 
-   b. Click **Add** to the right of **Command** to show the **Command** field then click in that field and select **Install ffmpeg** to automatically install the FFmpeg utility when the container starts.
+If you configured a certificate, select it in **Certificate ID**. A certificate is required if you want to select an external port other than the default **30027**.
 
-   {{< trueimage src="/images/SCALE/Apps/InstallNextcloudConfigAddInstallffmpetCommand.png" alt="Add Install ffmpeg Command" id="Add Install ffmpeg Command" >}}
+Enter the storage settings for each of the datasets created for the Nextcloud app.
+{{< expand "Configuring Nextcloud Storage" "v" >}}
+Do not select **DEPRECATED: Old Storage Structure** if you are deploying Nextcloud for the first time as this slows down the installation and is not necessary.
+If you are upgrading where your Nextcloud deployment in TrueNAS was a 1.x.x release, select this option.
+  
+Set **Host Path (Path that already exists on the system)** in **Type** for **Nextcloud AppData Storage**.
+Select **Enable ACL**, and then either enter or browse to and select the **html** dataset to populate the **Host Path** field.
 
-   c. (Optional) Click in the **Certificate Configuration** field and select the certificate for Nextcloud if already created and using a certificate.
+{{< trueimage src="/images/SCALE/Apps/InstallNextcloudStorageAppDataACLandACESettings.png" alt="Add Nextcloud Storage for AppData" id="Add Nextcloud Storage for AppData" >}}
 
-   d. Leave **Cronjobs** selected which enables this by default. Select the schedule you want to use for the cron job option.
+Select **Add** to the right of **ACL Entries**, add the **1001** user, and give it **FULL_CONTROL Access**.
```

Comment:
1001 is not correct here.
Nextcloud runs as root, but the subprocess of nextcloud runs as `33/www-data` 
So either no ACLs are needed, or use 33/www-data 
(nextcloud will _attempt_ to fix the permissions itself on the first install in either case. we dont control that)

Reply:
About user ID **1001** in the Nextcloud tutorial, I found this UID in the questions.yml file and using it does not prevent deploying the app. Should I say to enter this or **0** for root user? We don't enter the string the way you provided it 33/www-data, so I guess it would be just **33** entered as the ACE entry UID.  I can also provide the instructions to add this user from the Edit ACL screen like it is for the postgres_data dataset.

Reply:
Its possible 1001 was a typo. But its no longer there on docker apps!

33 is better yes, tho nextcloud will still attempt to fix perms on install.

Reply:
changed 1001 to 33